pilot/pkg/networking/util/util.go

```diff
@@ -276,6 +276,11 @@ func IsXDSMarshalingToAnyEnabled(node *model.Proxy) bool {
 	return !features.DisableXDSMarshalingToAny
 }
 
+// IsProtocolSniffingEnabled checks whether protocol sniffing is enabled.
+func IsProtocolSniffingEnabled(node *model.Proxy) bool {
```

Comment:
nit: see my earlier comment. You could rename this to `IsProtocolSniffingEnabledForNode` and then have a second method here `IsProtocolSniffingEnabledForPort`, which calls the first method and also checks that the port protocol is unsupported.